### ISSUE TITLE
[next] Tag other objects besides commits

### ISSUE THREAD

**Issue by clns** [libgit2/git2go#235]
Currently it's impossible to tag other git objects besides commits, since the API expects a commit object:

``` go
Create(name string, commit *Commit, tagger *Signature, message string) (*Oid, error)
CreateLightweight(name string, commit *Commit, force bool) (*Oid, error)
```

I think the API should be changed to accept an `Object` instead, following libgit2's pattern.

I can create a PR but I wanted to discuss it first. What do you think?

**Comment by carlosmn** [comment]
The fundamental problem here is that `Object` is an interface, and taking that in does not guarantee that you will see a libgit2 object so in order to tag any object in a way that doesn't involve guessing at the actual type, we'd have to export `gitObject`.

I've been unhappy with the interface for a while now. It works fine for taking stuff out of git2go, but putting stuff back in is annoying. I'm playing around with making `Object` a type, which is more in line with what it should be, since it's about what the objects are rather than what methods they provide.

**Comment by carlosmn** [comment]
The changes have landed on `next` so it should be possible to make `Describe()` apply to any object.

**Comment by carlosmn** [comment]
Fixed by #397
